### PR TITLE
Remove codec registry for inject/extract

### PR DESCRIPTION
**Pull request by tylerbenson**
Split inject/extract responsibility to different objects

In preparation for OT 0.32 support.

### CODE REVIEW

File: dd-trace-ot/src/main/java/datadog/opentracing/DDTracer.java [hunk 1]
```diff
@@ -332,24 +334,21 @@ public DDSpanBuilder buildSpan(final String operationName) {
 
   @Override
   public <T> void inject(final SpanContext spanContext, final Format<T> format, final T carrier) {
-
-    final Codec<T> codec = registry.get(format);
-    if (codec == null) {
+    if (!(carrier instanceof TextMap)) {
```
Comment (mar-kolya) [inline dd-trace-ot/src/main/java/datadog/opentracing/DDTracer.java]: Could you please 'swap' condition here so 'positive' result is first?

File: dd-trace-ot/src/main/java/datadog/opentracing/propagation/DatadogHttpCodec.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        carrier.put(SAMPLING_PRIORITY_KEY, String.valueOf(context.getSamplingPriority()));
+      }
+
+      for (final Map.Entry<String, String> entry : context.baggageItems()) {
+        carrier.put(OT_BAGGAGE_PREFIX + entry.getKey(), encode(entry.getValue()));
+      }
+      log.debug("{} - Parent context injected", context.getTraceId());
+    }
+
+    private String encode(final String value) {
```
Comment (mar-kolya) [inline dd-trace-ot/src/main/java/datadog/opentracing/propagation/DatadogHttpCodec.java]: Could you please add a comment why these values are url encoded/decoded?

Comment (tylerbenson) [reply dd-trace-ot/src/main/java/datadog/opentracing/propagation/DatadogHttpCodec.java]: I actually don't know... I think this was done in the original POC before my time.

Comment (mar-kolya) [reply dd-trace-ot/src/main/java/datadog/opentracing/propagation/DatadogHttpCodec.java]: :magic: :)

**Pull request merged**